apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -1087,6 +1087,59 @@ function handleAddPost( calypsoPort ) {
 	handleWpAdminRedirect( { calypsoPort, path: 'post-new.php', title: __( 'Add new post' ) } );
 }
 
+/**
+ * @external HTMLDivElement
+ */
+
+/**
+ * Extracts the post ID from the data-value attribute of the "Single Posts" command item,
+ * and encodes it for use in a URL.
+ * Example: data-value="Single Posts pub/fewer//single" -> "pub/fewer//single"
+ * Example: data-value="Single Posts pub/twentytwentyfour//single" -> "pub/twentytwentyfour//single"
+ * @param {HTMLDivElement} commandItem The command item element.
+ * @returns string
+ */
+function extractPostIdFromDataValueAttribute( commandItem ) {
+	const value = commandItem.getAttribute( 'data-value' );
+	const theme = value.replace( __( 'Single Posts' ), '' ).trim();
+
+	return encodeURIComponent( theme );
+}
+
+function handleSinglePosts( calypsoPort ) {
+	const selector = `[data-value^="${ __( 'Single Posts' ) }"]`;
+
+	const postMessage = ( commandItem ) => {
+		const postId = extractPostIdFromDataValueAttribute( commandItem );
```

Comment:
Or maybe check the commandItem here?

Reply:
It shouldn't happen, but I've added checks on https://github.com/Automattic/wp-calypso/pull/95432/commits/3c3ae2dcc97334be1826b4b0965b206939dd3c7b